### PR TITLE
[luci] Introduce fuse_rsqrt pass

### PR DESCRIPTION
**Pull request by jinevening**
This adds a new pass FuseRsqrtPass.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/12987
Draft PR: #12998

### CODE REVIEW

**Comment by seanshpark** [comment]
plz split PRs for different modules.

**Comment by jinevening** [comment]
Sorry. It was my mistake. I've divided source codes.

File: compiler/luci/pass/src/FuseRsqrtPass.cpp [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    {
+      case loco::DataType::S16:
+        CHECK_OR_FALSE(_div_const->quantparam() != nullptr);
+        CHECK_OR_FALSE(_div_const->quantparam()->scale.size() == 1);
+        CHECK_OR_FALSE(_div_const->quantparam()->zerop.size() == 1);
+        CHECK_OR_FALSE(_div_const->quantparam()->zerop.at(0) == 0);
+        CHECK_OR_FALSE(_div_const->size<loco::DataType::S16>() == 1);
+        CHECK_OR_FALSE(same(
+          1.0,
+          _div_const->at<loco::DataType::S16>(0) * _div_const->quantparam()->scale.at(0) == 1));
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseRsqrtPass.cpp:65]: there's `== 1` at the end, which makes this a bool...
is this intended?

Comment (jinevening) [reply compiler/luci/pass/src/FuseRsqrtPass.cpp]: No. It works as I expected by coincidence.. I've updated the code. Thanks for finding!

File: compiler/luci/pass/src/FuseRsqrtPass.cpp [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+  FuseRsqrt(const RsqrtPattern *p) : _p(p) {}
+
+public:
+  void apply(void);
+
+private:
+  luci::CircleRsqrt *create_rsqrt(loco::Graph *graph);
+
+private:
+  const RsqrtPattern *_p;
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseRsqrtPass.cpp:94]:
```suggestion
  const RsqrtPattern *_p = nullptr;
```

**Pull request merged**